OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs

```diff
@@ -143,17 +151,50 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
 			}
 			else
 			{
-				var aircraftInfo = transport.Info.TraitInfoOrDefault<AircraftInfo>();
-				if (aircraftInfo != null)
+				var aircraft = transport.TraitOrDefault<Aircraft>();
+				if (aircraft != null)
 				{
-					if (aircraftInfo.VTOL)
+					var destination = entryPath.Last();
+
+					// Try to find an alternative landing spot if we can't land at the current destination
+					if (!aircraft.CanLand(destination) && dropRange > 0)
 					{
-						transport.QueueActivity(new Turn(transport, aircraftInfo.InitialFacing));
+						foreach (var c in transport.World.Map.FindTilesInCircle(destination, dropRange))
+						{
+							if (!aircraft.CanLand(c))
+								continue;
+
+							if (cargo != null)
+							{
+								var mobiles = cargo.Passengers.Select(a =>
```

Comment:
Please pull this outside the `FindTilesInCircle` so that it isn't unnecessarily repeated.

Reply:
Something like the following should work:
```diff
--- a/OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs
+++ b/OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs
@@ -159,25 +159,22 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
                                        // Try to find an alternative landing spot if we can't land at the current destination
                                        if (!aircraft.CanLand(destination) && dropRange > 0)
                                        {
+                                               var mobiles = cargo != null ? cargo.Passengers.Select(a =>
+                                               {
+                                                       var mobileInfo = a.Info.TraitInfoOrDefault<MobileInfo>();
+                                                       if (mobileInfo == null)
+                                                               return new Pair<MobileInfo, uint>(null, 0);
+
+                                                       return new Pair<MobileInfo, uint>(mobileInfo, (uint)mobileInfo.GetMovementClass(a.World.Map.Rules.TileSet));
+                                               }) : new Pair<MobileInfo, uint>[0];
+
                                                foreach (var c in transport.World.Map.FindTilesInCircle(destination, dropRange))
                                                {
                                                        if (!aircraft.CanLand(c))
                                                                continue;
 
-                                                       if (cargo != null)
-                                                       {
-                                                               var mobiles = cargo.Passengers.Select(a =>
-                                                               {
-                                                                       var mobileInfo = a.Info.TraitInfoOrDefault<MobileInfo>();
-                                                                       if (mobileInfo == null)
-                                                                               return new Pair<MobileInfo, uint>(null, 0);
-
-                                                                       return new Pair<MobileInfo, uint>(mobileInfo, (uint)mobileInfo.GetMovementClass(a.World.Map.Rules.TileSet));
-                                                               });
-
-                                                               if (!mobiles.All(m => m.First == null || domainIndex.IsPassable(destination, c, m.First, m.Second)))
-                                                                       continue;
-                                                       }
+                                                       if (!mobiles.All(m => m.First == null || domainIndex.IsPassable(destination, c, m.First, m.Second)))
+                                                               continue;
 
                                                        destination = c;
                                                        break;
```